src/add-bibtex-reference.command.ts

```diff
@@ -0,0 +1,65 @@
+import joplin from "api";
+import constants from "./constants";
+import { showCitationPopup } from './ui/citation-popup';
+import { Reference } from "./model/reference.model";
+import { parse } from "./util/parser.util";
+import { DataStore } from "./data/data-store";
+const fs = joplin.require("fs-extra");
+
+const showMessageBox = joplin.views.dialogs.showMessageBox;
+
+/**
+ * Register the main command of the plugin
+ */
+export async function registerAddBibTexReferenceCommand () {
+    await joplin.commands.register({
+        name: constants.ADD_BIBTEX_REFERENCE_COMMAND,
+        label: 'Add BibTeX Reference',
+        iconName: constants.PLUGIN_ICON,
+        execute: async () => {
+
+            // Get file Path and check for errors
+            const filePath: string = await joplin.settings.value(constants.SETTINGS_FILE_PATH_ID);
+            if (!isFileExisting(filePath)) {
+                await showMessageBox(constants.ERROR_FILE_NOT_FOUND);
+                return;
+            }
+            if (!isFileReadable(filePath)) {
+                await showMessageBox(constants.ERROR_PERMISSION_DENIED);
+                return;
+            }            
```

Comment:
you don't need any of these checks. Just move the `readFile` call in the try/catch block (you should do that anyway), and if there's any error you'll catch it there.

Reply:
I thought this might be useful for showing descriptive error messages for the user. Am I missing something?

Reply:
You don't need that level of detail for something that's very unlikely to happen. But even then you can always give a descriptive error message in the catch block - `Could not open file: ${filePath}: ${error.message}`